### PR TITLE
Feature/grids

### PR DESCRIPTION
**Pull request by alex-outdoor**
Made a PR to make feedback easier but work not done. 
If I could have your comments on 2 issues I am facing : 

1) Some labels at some zoom level doesn't appeared to be centered inside the UTM zone. At some zoom level, it even appears twice and sometimes it's broken. 

2) At zoom level 11 for example, it takes longer to display the labels as it's looping a lot. Would there be a better way to optimize the computation ?

### CODE REVIEW

**Comment by JesseCrocker** [comment]
The only thing i noticed that seemed wrong from a quick look at the code is that the loop that adds the labels, line 842, is inside the loop thats started on line 769, Maybe thats causing the duplicated labels?

**Comment by alex-outdoor** [comment]
duplicatelabels
I have this issue where even if I check before drawing if the labelPt coordinate is inside the map screen - It still draws way too many points. Like on the pictures, it's supposed to show 12 waypoints but instead 156 points are drawn (and not all of them are visible which is weird).
I did try to console.log almost everything I could to do the math but didn't find what causes it.

**Comment by alex-outdoor** [comment]
I think this is ready to get reviewed. 
Next step : 
1) Integrate it to gaiacloud /map
2) Integrate it to the map print

**Comment by JesseCrocker** [comment]
@alex-outdoor I changed the base on this master, because the whole thing needs to be reviewed, not just your changes

**Comment by JesseCrocker** [comment]
There's a file that looks like it doesnt belong DuplicateLabels.jpg

**Comment by JesseCrocker** [comment]
It looks like maybe a rounding issue with the labels for the dms grid at zoom 15. the labels say 59" and 29" instead of 0" and 30"
screen shot 2016-12-09 at 6 57 09 am

**Comment by JesseCrocker** [comment]
I think the Grid spacing for MGRS might need to be shifted by 1 level, at zooms 14 and 17 I only get one or 2 squares on screen.

File: leaflet-grids.js [hunk 1]
```diff
@@ -0,0 +1,916 @@
...
+    
+    _gridSpacing: function () {
+        var zoom = this._map.getZoom();
+        if (zoom > 18) {zoom = 18}
+        return this.options.coordinateGridSpacing[zoom];
+    },
+        
+    _gridLines: function () {
+        var zoom = this._mapZoom; 
+        if(zoom <3){
```
Comment (JesseCrocker) [inline leaflet-grids.js]: Unless there is a good reason for this, i think at least the lat/lon grids should work down to zoom 0.

File: leaflet-grids.js [hunk 2]
```diff
@@ -0,0 +1,916 @@
...
+        var lines = [];
+        var zoom = this._map.getZoom();
+        if (zoom < 3){
+            return null;
+        }
+        var metersAtEquator = metersPerPixel(0, zoom);
+        var metersAtLat = metersPerPixel(this._map.getCenter().lat, zoom);
+        var gridSize = this._gridSize * metersAtEquator / metersAtLat;
+
+        console.log(this.gridLabel);
```
Comment (JesseCrocker) [inline leaflet-grids.js]: remove log

File: leaflet-grids.js [hunk 3]
```diff
@@ -0,0 +1,916 @@
...
+                        this._gridLabels.push(this._label(labelPt, gridLabel));
+                    }        
+                }
+            }
+        }
+
+        return lines;
+    },
+
+    _cleanHorz: function (line, leftLng, rightLng) {
```
Comment (JesseCrocker) [inline leaflet-grids.js]: Can you add a comment on what these _clean functions do?

**Review by JesseCrocker — CHANGES_REQUESTED** [review]
Code looks good, and it works great. Just need a couple minor changes.

**Comment by JesseCrocker** [comment]
This closes #1

**Pull request merged**